packages/contracts-bridge/contracts/BridgeRouter.sol

```diff
@@ -287,6 +323,45 @@ contract BridgeRouter is Version0, Router {
         _dust(_recipient);
     }
 
+    /**
+     * @notice Handles an incoming TransferToHook message.
+     *
+     * Tokens are sent to the Hook contract
+     * then IBridgeHook.onReceive callback is called
+     *
+     * @param _origin The domain of the chain from which the transfer originated
+     * @param _nonce The unique identifier for the message from origin to destination
+     * @param _tokenId The token ID
+     * @param _action The action
+     */
+    function _handleTransferToHook(
+        uint32 _origin,
+        uint32 _nonce,
+        bytes29 _tokenId,
+        bytes29 _action
+    ) internal {
+        // tokens will be sent to user-specified hook
+        address _hook = _action.evmHook();
+        // send tokens
+        address _token = _creditTokens(
+            _origin,
+            _nonce,
+            _tokenId,
+            _action,
+            _hook
+        );
+        bytes memory _call = abi.encodeWithSelector(
+            IBridgeHook.onReceive.selector,
+            _origin,
+            _tokenId.domain(),
+            _tokenId.id(),
+            _token,
+            _action.amnt(),
+            _action.extraData().clone()
```

Comment:
does this work if `extraData` is empty?

Reply:
yes, clone will produce a valid, empty, bytes mem